apps/aehyperchains/src/aehc_consensus_hyperchains.erl

```diff
@@ -497,16 +486,19 @@ state_pre_transform_micro_node(_Node, _PrevNode, _PrevKeyNode, Trees) -> Trees.
 
 %% -------------------------------------------------------------------
 %% Block rewards
+-spec state_grant_reward(pubkey(), trees(), amount()) -> trees().
 state_grant_reward(Beneficiary, Trees, Amount) -> %% TODO: we need the header here
-    M = fallback_consensus(),
-    M:state_grant_reward(Beneficiary, Trees, Amount).
+    apply_fallback(state_grant_reward, [Beneficiary, Trees, Amount]).
 
 %% -------------------------------------------------------------------
 %% PoGF
-pogf_detected(_H1, _H2) -> ok. %% TODO: we can't punish for forks due to forking on the parent chain - inspect the key headers at the given height
+pogf_detected(_H1, _H2) ->
+    ok. %% TODO: we can't punish for forks due to forking on the parent chain - inspect the key headers at the given height
 
 %% -------------------------------------------------------------------
 %% Genesis block
+-spec genesis_transform_trees(trees(), map()) -> trees() | no_return().
+-dialyzer({nowarn_function, genesis_transform_trees/2}).
```

Comment:
Hey, this is cheating :P Could you maybe add a comment why this dialyzer error can't be resolved?